File: 12-integer-to-roman/integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        h = [
           [ 'I', 1],
           [ 'IV', 4],
           [ 'V', 5],
           [ 'IX', 9],
           [ 'X', 10],
           [ 'XL', 40],
           [ 'L', 50],
           [ 'XC', 90],
            ['C', 100],
            ['CD', 400],
            ['D', 500],
            ['CM', 900],
            ['M', 1000],
        ]
        h.reverse()
        res = ''
        for key, n in h:
            count = num // n
            res += count * key
            num %= n
        return res

File: 12-integer-to-roman/integer-to-roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    const hashmap = {
        1:    'I',
        4:    'IV',
        5:    'V',
        9:    'IX',
        10:    'X',
        40:    'XL',
        50:    'L',
        90:    'XC',
        100:    'C',
        400:    'CD',
        500:    'D',
        900:    'CM',
        1000:    'M'
    }
    const intRoman = Object.keys(hashmap).map(e => Number(e))
    intRoman.sort((a, b) => a < b).reverse()
    let res = ''
    for (const int of intRoman) {
        const counts = Math.floor(num / int)
        res += hashmap[int].repeat(counts)
        num %= int
    }
    return res
};